sklearn/metrics/_scorer.py

```diff
@@ -669,21 +620,14 @@ def _get_response_method_name(response_method):
             list,
             tuple,
             StrOptions({"predict", "predict_proba", "decision_function"}),
+            Hidden(StrOptions({"default"})),
```

Comment:
Shall we have `None` as a deprecated `Options`?

Reply:
It's possible but I'm not sure it's worth. First the constraint would look like 
`Options(type=type(None), options={None}, deprecated={None})`
and mainly it makes the error message very unfriendly
`'a NoneType among {None (deprecated)}'`

Reply:
OK. so let's keep it as is.